substrate/client/rpc-spec-v2/src/common/events.rs

```diff
@@ -105,23 +109,41 @@ pub struct StorageResultErr {
 
 /// The result of a storage call.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-#[serde(untagged)]
-pub enum ArchiveStorageResult {
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "event")]
+pub enum ArchiveStorageEvent {
 	/// Query generated a result.
-	Ok(ArchiveStorageMethodOk),
+	Storage(StorageResult),
 	/// Query encountered an error.
-	Err(ArchiveStorageMethodErr),
+	StorageErr(ArchiveStorageMethodErr),
```

Comment:
I think this enum should be called `StorageError` to align with the spec (which uses the events "storage", "storageError" and "storageDone"